playerctl/playerctl-cli.c

```diff
@@ -294,10 +276,28 @@ static gboolean handle_player_command (gchar **command, PlayerctlPlayer *player,
   return FALSE;
 }
 
+static gboolean list_all_players(GError **error) {
```

Comment:
The control flow here is way too nested. You have two functions here with nearly identical names (`list_all_players()` and `list_player_names()`). Complicating things further is that there is a variable called `list_all_players_and_exit`. It's not at all clear what these things are doing just by glancing at the names, which is not quality code.